lte/gateway/python/scripts/dhcp_helper_cli.py

```diff
@@ -0,0 +1,353 @@
+        self._lease_expiration_time = None
+        self._iface = iface
+        self._mac = mac
+        self._vlan = vlan
+        self._ip = ip
+        self._server_ip = server_ip
+        self._pkt_queue: Queue = Queue()
+        self._ip_subnet = ""
+
+        self._sniffer = AsyncSniffer(
+            iface=iface,
+            filter=(self._SNIFFER_FILTER),
+            store=False,
+            prn=self._receive_answer,
+        )
+
+        self._sniffer.start()
+
+        time.sleep(self._SNIFFER_STARTUP_WAIT)
+
+    def allocate(self):
+        self.send_dhcp_discover()
+        self.wait_for(self.receive_dhcp_offer)
+        self.send_dhcp_request()
+        self.wait_for(self.receive_dhcp_ack)
+
+    def release(self):
+        self.send_dhcp_release()
+        self.wait_for(self.receive_dhcp_release)
+
+    def renew(self):
+        self._state = DHCPState.OFFER
+        self.send_dhcp_request()
+        self.wait_for(self.receive_dhcp_ack)
+
+    def _receive_answer(self, pkt: scapy.packet.Packet) -> None:
+        if DHCP in pkt:
+            self._pkt_queue.put(pkt)
+
+    def get_next_xid(self) -> int:
+        self._msg_xid += 1
+        return self._msg_xid
+
+    @staticmethod
+    def _get_option(packet: scapy.packet.Packet, name: str) -> Optional[str]:
+        for opt in packet[DHCP].options:
+            if opt[0] == name:
+                return opt[1]
+        return None
+
+    def send_dhcp_discover(self) -> None:
+        if self._state == DHCPState.DISCOVER:
+            dhcp_opts = [
+                ("message-type", "discover"),
+                "end",
+            ]
+            pkt_xid = self.get_next_xid()
+        else:
+            print(f"Wrong previous state {DHCPState(self._state).name} != DISCOVER")
+            return
+
+        self.send_dhcp_pkt(dhcp_opts, pkt_xid)
+
+    def send_dhcp_request(self) -> None:
+        if self._state == DHCPState.OFFER:
+            self._state = DHCPState.REQUEST
+            dhcp_opts = [
+                ("message-type", "request"),
+                ("requested_addr", self._ip),
+                ("server_id", self._server_ip),
+                "end",
+            ]
+            pkt_xid = self.get_next_xid()
+        else:
+            print(f"Wrong previous state {DHCPState(self._state).name} != OFFER")
+            return
+
+        self.send_dhcp_pkt(dhcp_opts, pkt_xid)
+
+    def send_dhcp_release(self) -> None:
+        self._state = DHCPState.RELEASE
+        dhcp_opts = [
+            ("message-type", "release"),
+            ("server_id", self._server_ip),
+            "end",
+        ]
+        pkt_xid = self.get_next_xid()
+        ciaddr = self._ip
+
+        self.send_dhcp_pkt(dhcp_opts, pkt_xid, ciaddr)
+
+    def send_dhcp_pkt(self, dhcp_opts: List[Any], pkt_xid: int, ciaddr: Optional[str] = None) -> None:
+        pkt = Ether(src=str(self._mac), dst="ff:ff:ff:ff:ff:ff")
+        if self._vlan and self._vlan != 0:
```

Comment:
`self._vlan` is int, so this may be a double check for != 0.

Reply:
How much do we trust those type annotations.